File: src/main/java/no/novari/instance/gateway/model/acos/AcosDocument.java
package no.novari.instance.gateway.model.acos;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.extern.jackson.Jacksonized;
import no.novari.flyt.instance.gateway.validation.constraints.ValidBase64;
import org.springframework.http.MediaType;

@Getter
@EqualsAndHashCode
@Jacksonized
@Builder
public class AcosDocument {

    @NotBlank
    private String name;

    @NotNull
    private MediaType type;

    private String encoding;

    @NotNull
    @ValidBase64
    private String base64;
}

File: src/main/kotlin/no/novari/flyt/acos/instance/gateway/model/acos/AcosDocument.kt
package no.novari.flyt.acos.instance.gateway.model.acos

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import no.novari.flyt.gateway.instance.validation.constraints.ValidBase64
import org.springframework.http.MediaType

data class AcosDocument(
    @field:NotBlank
    val name: String,
    @field:NotNull
    val type: MediaType,
    val encoding: String? = null,
    @field:NotNull
    @field:ValidBase64
    val base64: String,
)
